Lib/test/test_import/__init__.py

```diff
@@ -790,6 +790,70 @@ def test_issue105979(self):
         self.assertIn("Frozen object named 'x' is invalid",
                       str(cm.exception))
 
+    def test_cwd_script_shadowing_stdlib(self):
+        with CleanImport('collections'):
+            import collections
+            collections.__spec__ = types.SimpleNamespace()
+            collections.__spec__.origin = os.path.join(os.getcwd(), 'collections.py')
+
+            with self.assertRaisesRegex(
+                AttributeError,
+                r"module 'collections' has no attribute 'does_not_exist' \(most "
+                r"likely due to '.*collections.py' shadowing the standard "
+                r"library module named 'collections'\)"
+            ):
+                collections.does_not_exist
+
+            del collections.__spec__.origin
+            with self.assertRaisesRegex(
+                AttributeError,
+                r"module 'collections' has no attribute 'does_not_exist'$"
+            ):
+                collections.does_not_exist
+            del collections.__spec__
+            with self.assertRaisesRegex(
+                AttributeError,
+                r"module 'collections' has no attribute 'does_not_exist'$"
+            ):
+                collections.does_not_exist
+
+    def test_shadowing_stdlib_sys_edge_cases(self):
+        program = textwrap.dedent('''
+import collections
```

Comment:
Why dedent it if it's not indented in the first place. Either indent the whole inner text by 8 spaces or remove the dedent call.